graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+
+            scaled_df = scaled_df.withColumnRenamed(vector_col + "_scaled", vector_col)
+            scaled_df = scaled_df.select([vector_col] + other_cols)
+        elif shared_norm == "standard":
+            col_sums_df = vector_df.select(Summarizer.sum(vector_df[vector_col]).alias("sum"))
+
+            col_sums = col_sums_df.collect()[0]["sum"]  # type: DenseVector
+
+            col_sums_array = col_sums.toArray()
+            for i, col_sum in enumerate(col_sums_array):
+                if np.isinf(col_sum) or np.isnan(col_sum) or col_sum == 0:
+                    col_sums_array[i] = 0.0
+                else:
+                    col_sums_array[i] = 1.0 / col_sums_array[i]
+
+            elwise_divider = ElementwiseProduct(
+                scalingVec=DenseVector(col_sums_array),
+                inputCol=vector_col,
+                outputCol=f"{vector_col}_scaled",
+            )
+
+            scaled_df = elwise_divider.transform(vector_df).drop(vector_col)
+
+            scaled_df = scaled_df.withColumnRenamed(vector_col + "_scaled", vector_col)
+            scaled_df = scaled_df.select([vector_col] + other_cols)
+        else:
+            raise RuntimeError(f"Unknown normalizer requested for col {vector_col}: {shared_norm}")
+
+        return scaled_df
+
+    def apply(self, input_df: DataFrame) -> DataFrame:
+        def replace_empty_with_nan(x):
+            return F.when(x == "", "NaN").otherwise(x)
+
+        def convert_multistring_to_vector_df(
+            multi_string_df: DataFrame, separator: str
+        ) -> DataFrame:
+            """
+            Convert the provided DataFrame that is assumed to have one string
+            column named with the value of self.multi_column,
+            to a single-column DenseVector DF with the same column name.
+            """
+            # Hint: read the transformation comments inside-out, starting with split
+            vector_df = multi_string_df.select(
+                # transform outputs to an array so we convert to a DenseVector
+                array_to_vector(
+                    # After split, replace empty strings with 'NaN' and cast to Array<Float>
+                    F.transform(
+                        F.split(
+                            multi_string_df[self.multi_column], separator
+                        ),  # Split along the separator
+                        replace_empty_with_nan,
+                    ).cast(ArrayType(FloatType(), True))
+                ).alias(self.multi_column)
+            )
+            return vector_df
+
+        def vector_df_has_nan(vector_df: DataFrame, vector_col: str) -> bool:
+            """
+            Returns true if there exists at least one NaN value in `vector_df[vector_col]`
+            """
+            sum_vector_df = vector_df.select(Summarizer.mean(vector_df[vector_col]).alias("sum"))
+            sums_vector = sum_vector_df.take(1)[0]["sum"]
+            for val in sums_vector:
+                if np.isnan(val):
```

Comment:
Will an empty raw be treated as NaN?

Reply:
I believe so. The `Summarizer` will treat `None`/`NaN` the same I think, any such value in a column will result in a `Nan` being returned.

Reply:
Can you double check or write some unit test for it?